### PR TITLE
commit register

### PR DESCRIPTION
**Pull request by dqdung1230**

### CODE REVIEW

File: mock-project-backend/src/main/java/com/itsol/train/mock/dto/EmployeeDto.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Boolean isActived;
+    Date birthday;
+    String skypeAccount;
+    String email;
+
+    Boolean rememberMe;
+
+    Set<RoleEntity> roleEntities;
+
+    public EmployeeDto(EmployeeEntity employeeEntity) {
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/dto/EmployeeDto.java:49]: check employeeEntity null

File: mock-project-backend/src/main/java/com/itsol/train/mock/dto/ResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.itsol.train.mock.dto;
+
+import lombok.*;
+import lombok.experimental.FieldDefaults;
+
+@Getter
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/dto/ResponseDto.java]: Change getter & setter to @Data

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/DepartmentEntity.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @SequenceGenerator(name = "AUTO_INCRE_SEQ_DEPARTMENT", sequenceName = "AUTO_INCRE_SEQ_DEPARTMENT", allocationSize = 1)
+    Long id;
+
+    @Column(name = "name")
+    String name;
+
+    @Column(name = "location")
+    String location;
+
+//    public DepartmentEntity(Long id, Long managerId, String name, String location){
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/entity/DepartmentEntity.java]: remove comment

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/EmployeeIssueEntity.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@Entity
+@Table(name = "employee_issue")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+@FieldDefaults(level = AccessLevel.PRIVATE)
+
+public class EmployeeIssueEntity implements Serializable {
+
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/entity/EmployeeIssueEntity.java:17]: remove blank line

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/PositionEntity.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "AUTO_INCRE_SEQ_POSITION")
+    @SequenceGenerator(name = "AUTO_INCRE_SEQ_POSITION", sequenceName = "AUTO_INCRE_SEQ_POSITION", allocationSize = 1)
+    Long id;
+
+    @Column(name = "name")
+    String name;
+
+//    public PositionEntity (Long id, String name){}
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/entity/PositionEntity.java]: remove unused comment

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/ProjectEntity.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import java.io.Serializable;
+
+@Entity
+@Table(name = "project")
+@NoArgsConstructor
+@Data
+@AllArgsConstructor
+@ToString
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@EqualsAndHashCode
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/entity/ProjectEntity.java]: What is purpose here? If not, remove it

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+@Repository
+public class DepartmentRepositoryImpl implements DepartmentRepository {
+    public Logger logger = LoggerFactory.getLogger(DepartmentRepositoryImpl.class);
+
+    @Override
+    public DepartmentEntity findByDefault() {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            session.beginTransaction();
+            Query query = session.createQuery("select s from DepartmentEntity s where s.id = 0");
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java]: It should be 
`Query<DepartmentEntity> query = session.createQuery("select s from DepartmentEntity s where s.id = 0");`

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            DepartmentEntity singleResult = (DepartmentEntity) query.getSingleResult();
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java]: Don' cast type like this!

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/PositionRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+public class PositionRepositoryImpl implements PositionRepository {
+    public Logger logger = LoggerFactory.getLogger(PositionRepositoryImpl.class);
+
+    @Override
+    public PositionEntity findByDefault() {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+
+        try {
+            session.beginTransaction();
+            Query query = session.createQuery("select s from PositionEntity s where s.id = 0 ");
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/PositionRepositoryImpl.java]: It should be:
` Query<PositionEntity> query = session.createQuery("select s from PositionEntity s where s.id = 0 ");`

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/RoleRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+            session.beginTransaction();
+            Query<RoleEntity> query = session.createQuery("SELECT s FROM RoleEntity s WHERE s.id = 1");
+            RoleEntity singleResult = query.getSingleResult();
+            session.getTransaction().commit();
+            return singleResult;
+        } catch (Exception e) {
+            session.getTransaction().rollback();
+            logger.error(e.getMessage());
+        }
+//        finally {
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/RoleRepositoryImpl.java]: Finally should be used here!

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/TeamRepositoryImpl.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+@Repository
+public class TeamRepositoryImpl implements TeamRepository {
+    public Logger logger = LoggerFactory.getLogger(TeamRepositoryImpl.class);
+
+    @Override
+    public TeamEntity findByDefault() {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            session.beginTransaction();
+            Query query = session.createQuery("select s from TeamEntity s where s.id = 0");
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/TeamRepositoryImpl.java]: It should be:
`Query<TeamEntity> query = session.createQuery("select s from TeamEntity s where s.id = 0");`

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/EmployeeService.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+     * @param employeeDto UserDto
+     */
+    void register(EmployeeDto employeeDto) throws UsernameExistException, EmailExistException;
+
+    /**
+     *
+     * @return Optional<User>
+     */
+    Optional<EmployeeEntity> getEmployeeWithRoles();
+    boolean resetPassword(String email) throws UsernameNotFoundException;
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/service/EmployeeService.java:26]: Should insert a blank line above this line

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+        if (checkUsername.isPresent()) {
+            throw new UsernameExistException("Username has exist in database");
+        }
+        Optional<EmployeeEntity> checkEmail = employeeRepository.findOneWithAuthoritiesByEmail(username);
+        if (checkEmail.isPresent()) {
+            throw new EmailExistException("Email has exist in database");
+        }
+        EmployeeEntity entity = modelMapper.map(employeeDto, EmployeeEntity.class);
+        entity.setPassword(passwordEncoder.encode(entity.getPassword()));
+        entity.setIsActived(Boolean.TRUE);
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java]: It should be FALSE!?

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    private ModelMapper modelMapper;
+
+    @Autowired
+    private RoleRepository roleRepository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    @Override
+    public void register(EmployeeDto employeeDto) throws UsernameExistException, EmailExistException {
```
Comment (truongbb) [inline mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java:49]: add @Transasctional (of Spring) here

**Review by truongbb — COMMENTED** [review]
add ./mvn/ into .gitignore file

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/DepartmentEntity.java (hunk 3 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/entity/DepartmentEntity.java]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/EmployeeIssueEntity.java (hunk 4 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/entity/EmployeeIssueEntity.java:17]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/PositionEntity.java (hunk 5 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/entity/PositionEntity.java]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/entity/ProjectEntity.java (hunk 6 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/entity/ProjectEntity.java]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java (hunk 7 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/DepartmentRepositoryImpl.java]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/RoleRepositoryImpl.java (hunk 10 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/repo/impl/RoleRepositoryImpl.java]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/EmployeeService.java (hunk 12 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/service/EmployeeService.java:26]: ok

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java (hunk 14 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java:49]: OK

File: mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java (hunk 13 above)
Comment (dqdung1230) [reply mock-project-backend/src/main/java/com/itsol/train/mock/service/impl/EmployeeServiceImpl.java]: ok

**Pull request merged**